src/main/java/org/broadinstitute/hellbender/utils/variant/writers/GVCFWriter.java

```diff
@@ -155,9 +155,10 @@ public boolean checkError() {
     protected VariantContext addHomRefSite(final VariantContext vc, final Genotype g) {
 
         if (nextAvailableStart != -1) {
-            // don't create blocks while the hom-ref site falls before nextAvailableStart (for deletions)
-            if (vc.getStart() <= nextAvailableStart && vc.getContig().equals(contigOfNextAvailableStart)) {
-                return null;
+            //there's a use case here related to ReblockGVCFs for overlapping deletions on different haplotypes
+            if ( vc.getStart() <= nextAvailableStart && vc.getContig().equals(contigOfNextAvailableStart) ) {
+                if (vc.getEnd() <= nextAvailableStart)
```

Comment:
Could you add a unit case that triggers the first condition but not the second one?

Reply:
Yep, I have new tests in GVCFWriterUnitTests.